datasource/common/src/main/java/org/opennms/oce/datasource/common/SituationBean.java

```diff
@@ -45,7 +47,7 @@
 
     private Severity severity = Severity.INDETERMINATE;
     private List<ResourceKey> resourceKeys = new ArrayList<>();
-    private Set<Alarm> alarms = new LinkedHashSet<>();
+    private Map<String, Alarm> alarms = new HashMap<>();
```

Comment:
We could use a `LinkedHashMap` here if we want to preserve the ordering that the original implementation did. Not sure if we care.

Could also make this reference final as with my suggestion below we will only ever be modifying its contents.